todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/domain.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * -\-\-
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.example.android.architecture.blueprints.todoapp.statistics
+
+import com.example.android.architecture.blueprints.todoapp.data.Task
+
+sealed class StatisticsState {
+
+    object Loading : StatisticsState()
+
+    data class Loaded(@get:JvmName("activeCount") val activeCount: Int, @get:JvmName("completedCount") val completedCount: Int) : StatisticsState()
+
+    object Failed : StatisticsState()
+
+    companion object {
```

Comment:
and these convenience methods can then just move to the test since you'll still need them. You'd also need to make a change to them and cast all returns to the super type, so
```
fun loading() = Loading as StatisticsState
```